packages/core/src/components/Header/Navigation/MenuBar/Bar.styles.tsx

```diff
@@ -14,69 +10,57 @@ const show = {
   backgroundColor: theme.header.secondLevelBackgroundColor,
 };
 
-export const hide = {
+const hide = {
   top: 0,
   transition: ["top"],
   boxShadow: "none",
   transitionDuration: "300ms",
 };
 
-export const MenuBarRoot = styled(
-  "div",
-  transientOptions
-)(
-  ({
-    $type,
-    $hidden,
-    $active,
-  }: {
-    $type: string;
-    $hidden?: boolean;
-    $active?: boolean;
-  }) => ({
+export const { staticClasses, useClasses } = createClasses("HvHeader-MenuBar", {
+  root: {
     left: 0,
     width: "100%",
     height: "100%",
     display: "flex",
     justifyContent: "center",
     alignItems: "center",
-    ...($type === "menubar" && {
-      position: "relative",
-      backgroundColor: theme.header.backgroundColor,
-    }),
-    ...($type === "menu" && {
-      position: "absolute",
-      zIndex: -2,
-      height: theme.header.secondLevelHeight,
-      backgroundColor: theme.header.secondLevelBackgroundColor,
-      "& li > div": {
-        marginTop: 0,
-      },
-    }),
-    ...($hidden && { ...hide }),
-    ...($active && { ...show }),
-  })
-);
-
-export const MenuBarUl = styled("ul")({
-  margin: 0,
-  padding: 0,
-  display: "inherit",
-  alignItems: "center",
-  height: "100%",
-  [`&:hover .${headerMenuBarClasses.active}`]: {
-    ...hide,
-  },
-  [`& li:hover > .${headerMenuBarClasses.hidden}`]: {
-    ...show,
   },
-
-  [`&:focus-within .${headerMenuBarClasses.active}`]: {
-    ...hide,
+  menu: {
+    position: "absolute",
     zIndex: -2,
+    height: theme.header.secondLevelHeight,
+    backgroundColor: theme.header.secondLevelBackgroundColor,
+    "& li > div": {
+      marginTop: 0,
+    },
   },
-  [`& li:focus-within > .${headerMenuBarClasses.hidden}`]: {
-    ...show,
-    zIndex: -1,
+  menubar: {
+    position: "relative",
+    backgroundColor: theme.header.backgroundColor,
+  },
+  hidden: { ...hide },
+  active: { ...show },
+  list: {
+    margin: 0,
+    padding: 0,
+    display: "inherit",
+    alignItems: "center",
+    height: "100%",
+    [`&:hover $active`]: {
```

Comment:
No need for the computed keys

```suggestion
    "&:hover $active": {
```

Reply:
Thanks, I didn't notice! I will update 🙏